src/mongo/client/command_writer.cpp:68:78: error: no member named 'next' in namespace 'boost'
            const std::vector<WriteOperation*>::const_iterator next = boost::next(batch_iter);
                                                                      ~~~~~~~^
src/mongo/util/net/message_port.h:34:38: error: expected class name
class AbstractMessagingPort : boost::noncopyable {
                                     ^

--- src/mongo/util/net/message_port.h.orig	2016-06-21 19:15:24 UTC
+++ src/mongo/util/net/message_port.h
@@ -20,8 +20,15 @@
 #include "mongo/config.h"
 
 #include <boost/shared_ptr.hpp>
-#include <boost/utility.hpp>
 #include <vector>
+
+#include <boost/version.hpp>
+#if BOOST_VERSION < 106700
+#include <boost/utility.hpp>
+#else
+#include <boost/next_prior.hpp>
+#include <boost/noncopyable.hpp>
+#endif
 
 #include "mongo/util/net/message.h"
 #include "mongo/util/net/sock.h"
